shared/subspace-data-retrieval/src/object_fetcher.rs

```diff
@@ -611,10 +611,10 @@ where
     /// The mapping is only used for error reporting.
     async fn read_pieces(
         &self,
-        piece_indexes: &Vec<PieceIndex>,
+        piece_indexes: Vec<PieceIndex>,
         mapping: GlobalObject,
     ) -> Result<Vec<Piece>, Error> {
-        download_pieces(piece_indexes, &self.piece_getter)
+        download_pieces(piece_indexes.clone(), &self.piece_getter)
```

Comment:
It is unfortunate that we have to clone it just to log in error case. I was using `Arc<[PieceIndex]>` in some places for this reason.